### PR TITLE
fix: Improved TextMate grammar tokenization

### PR DESCRIPTION
**Pull request by jirimosinger**

### CODE REVIEW

**Comment by jirimosinger** [comment]
@VitGottwald: Just FYI:

- I've looked into the old TextMate and I think the lookbehinds there shouldn't cause any problems when `joni` regex is used as the lookbehinds have only fixed lengths:
  - [Standard syntax](https://github.com/BroadcomMFD-DevOps/che-che4z-lsp-for-hlasm/blob/33f1f233a1e65ea73b30815529e00f0d4dd25b22/clients/vscode-hlasmplugin/syntaxes/hlasm.tmLanguage.json)
  - [Listing syntax](https://github.com/BroadcomMFD-DevOps/che-che4z-lsp-for-hlasm/blob/33f1f233a1e65ea73b30815529e00f0d4dd25b22/clients/vscode-hlasmplugin/syntaxes/hlasmListing.tmLanguage.json)

- This PR removes incorrect tokenization of spaces and more

**Comment by VitGottwald** [comment]
Thank you @jirimosinger.

Can you, please, share a link to the old grammar you were looking at (so that we can take the exact one and try it)?

File: clients/vscode-hlasmplugin/scripts/syntaxes/hlasm_tmgrammar_template.txt [hunk 1]
```diff
@@ -386,24 +412,51 @@
       "name": "constant.numeric.hlasm"
     },
     "variable": {
-      "match": "&(([A-Za-z@#$_][A-Za-z@#$_0-9]*)|([(].*?[)]))(?<=^${listingDetails}$.{0,71})",
-      "name": "variable.other.readwrite.hlasm"
+      "match": "(?:(?:&[A-Za-z@#$_][A-Za-z@#$_0-9]*)|(&[(].*?[)])([(].*?[)])?)(?<=^${listingDetails}$.{0,71})",
```
Comment (slavek-kucera) [inline clients/vscode-hlasmplugin/scripts/syntaxes/hlasm_tmgrammar_template.txt:415]: The capture groups seem off + the `&` should be kept outside?

File: clients/vscode-hlasmplugin/scripts/syntaxes/hlasm_tmgrammar_template.txt [hunk 2]
```diff
@@ -386,24 +412,51 @@
...
+      "captures": {
+        "1": {
+          "name": "keyword.other.hlasm"
+        }
+      }
     },
     "sequenceNumbers": {
-      "match": "\\s*.?(?<=^${listingDetails}$.{72}).+$",
-      "name": "invalid.ignored"
+      "match": "\\s*.?(?<=^${listingDetails}$.{72})\\s*(\\S(?:.*\\S)?)\\s*?$",
```
Comment (slavek-kucera) [inline clients/vscode-hlasmplugin/scripts/syntaxes/hlasm_tmgrammar_template.txt:454]: Everything after 72 is ignored (and corresponding stuff in listings) whether it is a space or not so I think just `(.+)` would be appropriate.

File: clients/vscode-hlasmplugin/scripts/syntaxes/hlasm_tmgrammar_template.txt (hunk 1 above)
Comment (slavek-kucera) [reply clients/vscode-hlasmplugin/scripts/syntaxes/hlasm_tmgrammar_template.txt:415]: I looked at it again and just the ampersand should be kept outside.

**Review by slavek-kucera — APPROVED** [review]

**Pull request merged**